bentoml/cli/deployment.py

```diff
@@ -147,341 +146,79 @@ def get_state_after_await_action_complete(
+        help='Wait for apply action to complete or encounter an error.'
+        'If set to no-wait, CLI will return immediately. The default value is wait',
     )
-    def create(
-        name,
-        bento,
-        platform,
-        output,
-        namespace,
-        labels,
-        annotations,
-        region,
-        instance_type,
-        instance_count,
-        num_of_gunicorn_workers_per_instance,
-        api_name,
-        kube_namespace,
-        replicas,
-        service_name,
-        service_type,
-        memory_size,
-        timeout,
-        wait,
-    ):
-        # converting platform parameter to DeploymentOperator name in proto
-        # e.g. 'aws-lambda' to 'AWS_LAMBDA'
-        track_cli('deploy-create', platform.replace('-', '_').upper())
-        bento_name, bento_version = bento.split(':')
-        operator_spec = {
-            'region': region,
-            'instance_type': instance_type,
-            'instance_count': instance_count,
-            'num_of_gunicorn_workers_per_instance': num_of_gunicorn_workers_per_instance,  # noqa E501
-            'api_name': api_name,
-            'kube_namespace': kube_namespace,
-            'replicas': replicas,
-            'service_name': service_name,
-            'service_type': service_type,
-            'memory_size': memory_size,
-            'timeout': timeout,
-        }
-        yatai_client = YataiClient()
+    def deploy(deployment_yaml, output, wait):
+        track_cli('deploy-deploy', deployment_yaml.get('spec', {}).get('operator'))
         try:
-            result = yatai_client.deployment.create(
-                name,
-                namespace,
-                bento_name,
-                bento_version,
-                platform,
-                operator_spec,
-                parse_key_value_pairs(labels),
-                parse_key_value_pairs(annotations),
-            )
-        except BentoMLException as e:
-            _echo(
-                'Failed to create deployment {}.: {}'.format(name, str(e)),
-                CLI_COLOR_ERROR,
-            )
-            return
-
-        if result.status.status_code != status_pb2.Status.OK:
-            _echo(
-                'Failed to create deployment {name}. {error_code}:'
-                '{error_message}'.format(
-                    name=name,
-                    error_code=status_pb2.Status.Code.Name(result.status.status_code),
-                    error_message=result.status.error_message,
-                ),
-                CLI_COLOR_ERROR,
-            )
-            return
-        else:
-            if wait:
-                result_state = get_state_after_await_action_complete(
-                    yatai_client=yatai_client,
-                    name=name,
-                    namespace=namespace,
-                    message='Creating deployment ',
+            yatai_client = YataiClient()
+            result = yatai_client.deployment.apply(deployment_yaml)
```

Comment:
let's add a `result = yatai_client.deployment.create(deployment_yaml)`, which should be different than `apply`.